sqlglot/transforms.py

```diff
@@ -9,6 +9,27 @@
     from sqlglot.generator import Generator
 
 
+def update_from(expression: exp.Expression) -> exp.Expression:
+    """Transform UPDATE FROM to MERGE INTO"""
+    from_expression = expression.find(exp.From)
+    if isinstance(expression, exp.Update) and from_expression:
+        join = from_expression.find(exp.Join)
+        if join:
```

Comment:
This looks incomplete, can't we have more than one joins? What happens in that case?

Reply:
If any join then follow that path. 
It is probably worth building more complex test cases to explore these cases.

Reply:
> If any join then follow that path.

You would ignore the rest of the joins though, since you only deal with the first one.

Reply:
Found real-world example and tested. The generated join is far from complete. Will have to research.

Reply:
The strategy is to convert the multiple joins into a sub-select...
```sql
-- tsql
UPDATE ru_new SET NPI = i.identifier
      FROM t_merge ru_new
      JOIN mytbl1 ui
        ON ru_new.user_id = ui.user_id
      JOIN mytbl2 i
        ON ui.identifier_id = i.id
      JOIN mytble3 it
        ON i.identifier_type = it.identifier_type
     WHERE it.short_description = 'NPI'
        and i.identifier like replicate('[0-9]', 10)

-- spark delta sql/databricks sql
MERGE INTO t_merge AS ru_new
USING (
    SELECT
        ui.user_id,
        i.identifier AS npi
    FROM mytbl1 ui
    JOIN mytbl2 i
        ON ui.identifier_id = i.id
    JOIN mytble3 it
        ON i.identifier_type = it.identifier_type
    WHERE it.short_description = 'NPI'
        AND i.identifier rLIKE '^[0-9]{10}$'
) AS src
ON ru_new.user_id = src.user_id
WHEN MATCHED THEN
    UPDATE SET ru_new.NPI = src.npi;
```

Reply:
I'll have to do some more work, thanks for the catch :)

Reply:
I've done the work for the test above. I haven't pushed it because it's ugly, lol.

Reply:
Can you share a link so that I can take a quick look and see if it's in the right direction?

Reply:
@GeorgeSittas If I push, it goes directly into this PR.

Reply:
@GeorgeSittas Here you go: https://gist.github.com/dmoore247/2d2feff1dd79c63d49fc00eb9892e1de

Reply:
Yeah that looks overly complex, can you briefly explain what you tried to implement?